assets/wire/tests/WireApp.spec.tsx

```diff
@@ -124,6 +125,20 @@ describe('shortHighlightedtext', () => {
         expect(output).toEqual(`On olemassa myös sellaisia ihmisiä, joiden silmissä
         jokainen parisuhde muuttuu <span class="es-highlight">mukaanalkuhuuma</span>  <span class="es-highlight">kestää</span> yleensä 1–2 vuotta...`);
     });
-       
+
+    it('returns highlight text condition with more tags', () => {
+        const html = '<ul><li><span class="es-highlight">foo</span></li><li> <span class="es-highlight">foo</span></li><li>bar</li><li> <span class="es-highlight">foo</span></li></ul>';
+        const maxLength = 40;
+        const output = shortHighlightedtext(html, maxLength);
+        expect(output).toEqual('<span class="es-highlight">foo</span> <span class="es-highlight">foo</span> <span class="es-highlight">foo</span>...');
+    });
+
+    it('returns highlight text condition with more matching words', () => {
+        const html = `<p>some <span class="es-highlight">matching</span> text and <span class="es-highlight">more</span> matching text.</p>
+        <p>another <span class="es-highlight">matching</span> text.</p>`;
+        const maxLength = 40;
+        const output = shortHighlightedtext(html, maxLength);
+        expect(output).toEqual('some <span class="es-highlight">matching</span> text and  text and <span class="es-highlight">more</span> matching text.another <span class="es-highlight">matching</span> text....');
```

Comment:
here we should make sure we're not repeating the `text and` that should be there only once. and would be good to add space after `text.`